NBitcoin/PubKey.cs

```diff
@@ -351,8 +351,9 @@ public override bool Equals(object obj)
 			PubKey item = obj as PubKey;
 			if(item == null)
 				return false;
-			return ToHex().Equals(item.ToHex());
+			return Equals(item);
 		}
+		public bool Equals(PubKey pk) => pk != null && Utils.ArrayEqual(ToBytes(), pk.ToBytes());

```

Comment:
No need to make a copy of the internal representation to compare them. Use simply the `vch` field or `.ToBytes(true)`